staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/metrics.go

```diff
@@ -58,3 +71,23 @@ func (c *certificateExpirationTracker) set(a *Authenticator, t time.Time) {
 		c.metricSet(&earliest)
 	}
 }
+
+// incrementCallsMetric increments a global metrics counter for the number of calls to an exec
+// plugin, partitioned by exit code. The provided err should be the return value from
+// exec.Cmd.Run().
+func incrementCallsMetric(err error) {
+	switch err.(type) {
+	case nil: // Binary execution succeeded.
+		metrics.ExecPluginCalls.Increment(0, pluginCalledStatus)
+
+	case *exec.ExitError: // Binary execution failed (see "os/exec".Cmd.Run()).
+		metrics.ExecPluginCalls.Increment(err.(*exec.ExitError).ExitCode(), pluginCalledStatus)
+
+	case *exec.Error: // Binary does not exist (see exec.Error).
+		metrics.ExecPluginCalls.Increment(0, pluginNotFoundStatus)
```

Comment:
suggest setting code to 1 for pluginNotFoundStatus and clientInternalError... that allows someone to monitor non-0 codes for all unsuccessful invocations, and partition by the event type to understand the reason for the failure

Reply:
2da2732ae9cd5e83b01fceb50ce1c4606e0add28